subsys/net/ip/Kconfig.tcp

```
@@ -230,8 +230,8 @@ config NET_TCP_ISN_RFC6528
 	default y
 	depends on NET_TCP
 	select MBEDTLS
-	select MBEDTLS_MD
-	select MBEDTLS_MAC_MD5_ENABLED
+	select MBEDTLS_PSA_CRYPTO_C
```

```suggestion
	select MBEDTLS_PSA_CRYPTO_C depens on !BUILD_WITH_TFM
```
Selecting MBEDTLS_PSA_CRYPTO_C is incompatible with BUILD_WITH_TFM. Not sure if this depends will help, but you could try it out